src/main/java/com/google/sps/servlets/AddSaleCardServlet.java

```diff
@@ -99,44 +107,75 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-      // TODO: Create vendor key using the Auth ID
-      // Key vendorKey = KeyFactory.createKey("Vendor", (String) user.authID);
-
-      // Generate Key when datastore put
-      Entity saleCard = new Entity("SaleCard"/*, vendorKey*/);
-      datastore.put(saleCard);      
-
-      // TODO: Remove from datastore existing saleCard, picture and location data
+      Key vendorKey = KeyFactory.createKey("Vendor", userService.getCurrentUser().getUserId());
+      Entity vendorEntity;
+      try {
+        vendorEntity = datastore.get(vendorKey);
+      } catch (EntityNotFoundException e) {
+        System.out.println(e);
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      Vendor vendorObject = new Vendor(vendorEntity);
+
+      Entity saleCard;
+      Entity picture;
+      Entity locationData;
```

Comment:
```suggestion
      Entity saleCard, picture, locationData;
```

Reply:
Thanks! Idk why my first thought was that Java wouldn't let me do that haha